### PR TITLE
FINALLY FIGURED THIS OUT

### PR DESCRIPTION
**Pull request by glenvollmer**
completed html and javascript for the christmas tree challenge

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		tree = space.repeat(inputHeight.value - i - 1) + inputCharacter.value.repeat((i * 2) + 1);
+		console.log(tree)
+	}
+});
+
+document.getElementById("button").addEventListener("click", function errorMessage() {
+	if (inputCharacter.value === "", inputHeight.value === "") {
+	document.getElementById("error").innerHTML = "Please Fill Out Fields";
+	}	
+})
```
Comment (JoeShep) [inline main.js:21]: There's no need to attach two event listeners to the button. With one click listener you can check for proper input then if it passes muster, you can run the tree building function.

File: main.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+
+
+
+
```
Comment (JoeShep) [inline main.js:4]: Try to avoid unnecessary blank lines

**Review by JoeShep — COMMENTED** [review]
Glen, this is a great start. The tree builds and looks as expected, even if I build multiple trees in a row. Nice.

But, you need to look over the app requirements and refactor this to  meet those requirements. First, note that it asks you to create a tree builder function that takes an object as an argument. That object should contain two properties: the tree height and character.

Secondly, this also needs to work when the user hits the return/enter key.

Let me know when you've updated this and I'll take a look.

**Comment by JoeShep** [comment]
Glen, as I mentioned the other day you should keep a PR open and continue to work on the same branch and push up your changes. Then when you get a thumbs-up you can merge. You can go ahead and make a new PR this time, but keep that in mind for next time.

**Comment by glenvollmer** [comment]
whoops! sorry about that. I definitely will do that from now on.

**Pull request merged**